From 31411ff5059f879728bbf0da94f7fbef89733ca3 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Tue, 22 Aug 2023 09:53:27 +0200
Subject: [PATCH 0430/1141] clk: stm32mp2: check the clock access from bus
 interface

Use the check interface to control if the clock is available
without taking the semaphore.

Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Change-Id: I6899bb46e20d189d8412d54ce29f0a282d5050bc
---
 drivers/clk/stm32/clk-stm32mp25.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/clk/stm32/clk-stm32mp25.c
+++ b/drivers/clk/stm32/clk-stm32mp25.c
@@ -988,7 +988,7 @@ static int stm32mp25_check_security(void
 		u32 index = (u32)cfg->sec_id;
 
 		if (index & SEC_RIFSC_FLAG)
-			ret = stm32_rifsc_get_access_by_id(index & ~SEC_RIFSC_FLAG);
+			ret = stm32_rifsc_check_access_by_id(index & ~SEC_RIFSC_FLAG);
 		else
 			ret = stm32_rcc_get_access(base, cfg->sec_id & ~SEC_RIFSC_FLAG);
 	}
